src/backend/gl/src/queue.rs

```diff
@@ -1195,6 +1211,14 @@ impl hal::queue::CommandQueue<Backend> for CommandQueue {
             }
         }
 
+        // Create a fence used to synchronize the finish of the rendering and
+        // the blit used to present on the surface.
+        self.presentation_fence = self
```

Comment:
We are creating the fence but never waiting on it. In present() there should be a call to `wait_sync` on that fence.

Reply:
Ah, right again. I think I took that out and forgot to put it back in.